go/api/base_client.go

```diff
@@ -199,3 +200,102 @@ func (client *baseClient) DecrBy(key string, amount int64) (int64, error) {
 	}
 	return handleLongResponse(result), nil
 }
+
+func (client *baseClient) HGet(key, field string) (string, error) {
+	result, err := client.executeCommand(C.HGet, []string{key, field})
+	if err != nil {
+		return "", err
+	}
+
+	return handleStringOrNullResponse(result), nil
+}
+
+func (client *baseClient) HGetAll(key string) (map[string]string, error) {
+	result, err := client.executeCommand(C.HGetAll, []string{key})
+	if err != nil {
+		return map[string]string{}, err
+	}
+
+	return handleStringMapResponse(result), nil
+}
+
+func (client *baseClient) HMGet(key string, fields []string) ([]string, error) {
+	result, err := client.executeCommand(C.HMGet, append([]string{key}, fields...))
+	if err != nil {
+		return []string{}, err
+	}
+
+	return handleStringArrayResponse(result), nil
+}
+
+func (client *baseClient) HSet(key string, fields map[string]string) (int64, error) {
+	result, err := client.executeCommand(C.HSet, utils.ConvertMapToKeyValueStringArray(key, fields))
+	if err != nil {
+		return 0, err
+	}
+
+	return handleIntegerOrNullResponse(result), nil
```

Comment:
`HSET` does not return null, I would expect here: `handleIntegerResponse`

If a null was returned, then we should return an error and not a default value

Reply:
@eifrah-aws the mentioned changes have been implemented.